components/Home/VeOlas/ManageYourLock/IncreaseUnlockTime.jsx

```diff
@@ -0,0 +1,78 @@
+import { useSelector, useDispatch } from 'react-redux';
+import { Button, Form, Typography } from 'antd/lib';
+import {
+  fetchMappedBalances,
+  fetchVotesAndTotalSupplyLocked,
+} from 'store/setup/actions';
+import {
+  notifyError,
+  notifySuccess,
+  CannotIncreaseAlert,
+} from 'common-util/functions';
+import { parseToSeconds, FormItemDate } from '../../common';
+import { updateIncreaseUnlockTime } from '../utils';
+
+const { Title } = Typography;
+
+export const IncreaseUnlockTime = () => {
+  const dispatch = useDispatch();
+  const account = useSelector((state) => state?.setup?.account);
+  const chainId = useSelector((state) => state?.setup?.chainId);
+  const cannotIncreaseTime = useSelector(
+    (state) => state?.setup?.mappedBalances?.isMappedAmountZero,
+  );
+  const mappedEndTime = useSelector(
+    (state) => state?.setup?.mappedBalances?.endTime || null,
+  );
+
+  const [form] = Form.useForm();
+
+  const onFinish = async (e) => {
+    try {
+      const txHash = await updateIncreaseUnlockTime({
```

Comment:
Where is this checked for the time to be not longer than the max possible one? It is checked on-chain, but I suppose it's checked off-chain as well?